base/inference.jl

```diff
@@ -2276,14 +2314,25 @@ function effect_free(e::ANY, sv, allow_volatile::Bool)
                 return false
             end
         end
-        return true
     end
-    return false
+    return true
```

Comment:
this adds control-flow to `effect-free`. while I agree that's essentially correct, I believe `inlineable` may be surprised by that and not work correctly (because control flow might not be effect-free for inlining)